sensor/common/registry/registry_store.go

```diff
@@ -122,12 +153,53 @@ func (rs *Store) GetRegistryForImage(image *storage.ImageName) (registryTypes.Re
 		}
 	}
 
-	return nil, errors.Errorf("Unknown image registry: %q", reg)
+	return nil, errors.Errorf("unknown image registry: %q", reg)
 }
 
-// HasRegistryForImage returns true when the registry store has the registry
-// for the given image.
-func (rs *Store) HasRegistryForImage(image *storage.ImageName) bool {
-	reg, err := rs.GetRegistryForImage(image)
-	return reg != nil && err == nil
+// getOrInitGlobalRegistries will return global registries if existing, or create a new set
+// mimics `getRegistries`
+func (rs *Store) getOrInitGlobalRegistries() registries.Set {
+	rs.mutex.Lock()
+	defer rs.mutex.Unlock()
+
+	if rs.globalRegistries == nil {
+		rs.globalRegistries = registries.NewSet(rs.factory)
+	}
+
+	return rs.globalRegistries
+}
+
+// UpsertGlobalRegistry will store a new registry with the given credentials into the global registry store
+func (rs *Store) UpsertGlobalRegistry(ctx context.Context, registry string, dce config.DockerConfigEntry) error {
```

Comment:
When would I use this vs `UpsertRegistry`?

Reply:
When the registry being inserted is not namespace bound, and can apply to any workload - the OpenShift global pull secret is the only scenario i'm aware of that this is needed for

Reply:
makes sense. Similar question for `GetGlobalRegistryForImage`. How would a user know when the registry they are looking for is global vs namespaced?

Reply:
I believe that depends on the use case / usage - in this case the only usage is in `sensor/common/scan/scan.go` - any matching global registry is added to the 'list' of registries to try